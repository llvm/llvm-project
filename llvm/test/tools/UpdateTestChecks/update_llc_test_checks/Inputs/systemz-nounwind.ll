; RUN: llc -mtriple=s390x-unknown-linux < %s | FileCheck %s

define i32 @unwind(i32 %a, i32 %b) {
  %add = add i32 %a, %b
  ret i32 %add
}

define i32 @nounwind(i32 %a, i32 %b) nounwind {
  %add = add i32 %a, %b
  ret i32 %add
}
