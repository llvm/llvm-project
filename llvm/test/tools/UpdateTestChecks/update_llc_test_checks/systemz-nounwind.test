# REQUIRES: systemz-registered-target
## Check that nounwind functions are handled.

# RUN: cp -f %S/Inputs/systemz-nounwind.ll %t.ll && %update_llc_test_checks %t.ll
# RUN: diff -u %S/Inputs/systemz-nounwind.ll.expected %t.ll
