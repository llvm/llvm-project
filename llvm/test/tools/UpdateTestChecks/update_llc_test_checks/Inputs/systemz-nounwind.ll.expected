; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
; RUN: llc -mtriple=s390x-unknown-linux < %s | FileCheck %s

define i32 @unwind(i32 %a, i32 %b) {
; CHECK-LABEL: unwind:
; CHECK:       # %bb.0:
; CHECK-NEXT:    ar %r2, %r3
; CHECK-NEXT:    br %r14
  %add = add i32 %a, %b
  ret i32 %add
}

define i32 @nounwind(i32 %a, i32 %b) nounwind {
; CHECK-LABEL: nounwind:
; CHECK:       # %bb.0:
; CHECK-NEXT:    ar %r2, %r3
; CHECK-NEXT:    br %r14
  %add = add i32 %a, %b
  ret i32 %add
}
